common/compare.py

```diff
@@ -0,0 +1,111 @@
+model/module. The input model/module is not stored in the database, it is 
+fitted to graph2vec with the models/modules present in the database and 
+cosine similarity is computed using embeddings.
+
+CLA to module and Args to functions in module:
+    file_path (str) : Full path to the model/module to be parsed.
+    file_format (str) : String to denote the file format of the input model/module.
+        One of "TFFrozenGraph", "TFSavedModel" or "TFLite".
+    model_name (str) : Name of the input model/module.
+    include_edge_attrs (str) : Case insensitive string to denote whether to 
+        include edge attributes of input edge from _EDGE_ATTRS in feature,
+        if "true" then they are included.
+    include_node_attrs (str) : Case insensitive string to denote whether to 
+        include node attributes from _NODE_ATTRS in feature, if "true" then 
+        they are included.
+    wl_iterations (int) : Depth of subgraph rooted at every node to be 
+        considered for feature building in graph2vec.
+"""
+
+from sklearn import metrics
+import argparse
+import numpy
+
+from common import Storage
+from common import Vectorize
+from TF.src import TFParser
+from TFLite.src import TFLiteParser
+
+def topk_similar_in_db(file_path, file_format, model_name, include_node_attrs,
+                        include_edge_attrs, topk):
+    """ Module to print topk similar models to a file
+
+    Takes an input model/module and a list of model_graphs present in the 
+    database, parses the file and adds its graph to the models graphs and
+    obtains embeddings for the collective list. Then prints the topk models
+    similar to the input model/module from the database models.
+    """
+
+    # Instance and database ID of spanner database which holds the models
+    INSTANCE_ID = 'ml-models-characterization-db'
+    DATABASE_ID = 'models_db'
+
+    # Parsing models from database into Graph objects
+    storage = Storage.Storage(INSTANCE_ID, DATABASE_ID)
+    model_graphs = storage.parse_models()
+
+    # Parsing file_path depending on file_format
+    if file_format == "TFFrozenGraph":
+        parser = TFParser.TFParser()
+        graph = parser.parse_graph(file_path, model_name, None, None, "False", [])
+
+    elif file_format == "TFSavedModel":
+        parser = TFParser.TFParser()
+        graph = parser.parse_graph(file_path, model_name, None, None, "True", [])
+
+    else:
+        parser = TFLiteParser.TFLiteParser()
+        graph = parser.parse_graph(file_path, model_name, None, None)
+
+    # Adding current graph to database models to fit to graph2vec
+    model_graphs.append(graph)
+
+    # Getting graph2vec embeddings
+    vectorize = Vectorize.Vectorize()
+    model_graphs, embeddings = vectorize.get_graph2vec_embeddings(
+        model_graphs, include_edge_attrs, include_node_attrs, wl_iterations)
+
+    # Cosine similarity
+    similarity = metrics.pairwise.cosine_similarity(
+        [embeddings[len(embeddings) - 1]], embeddings)
```

Comment:
Is it always guaranteed that get_graph2vec_embedding will not return empty 'embeddings'?

Reply:
Guaranteed that 'embeddings' will not be empty since 'model_graphs' will contain at least the file being parsed.
Even in other cases, empty 'embeddings' only possible on empty 'model_graphs' in which case graph2vec will throw an error.